crates/biome_service/src/workspace/server.rs

```diff
@@ -396,6 +396,11 @@ impl WorkspaceServer {
 
         let documents = self.documents.pin();
         documents.compute(path.clone(), |current| {
+            let biome_path = BiomePath::new(&path);
+            if biome_path.is_dependency() && biome_path.is_type_declaration() && current.is_none() {
```

Comment:
This may even be enough?

```suggestion
            if biome_path.is_dependency() && current.is_none() {
```

That way you don't keep the content for `package.json` files of dependencies either.

And I'm even tempted to say, can't we just do this instead:

```suggestion
            if opened_by_scanner && current.is_none() {
```

Reply:
I tried the second suggestion, and unfortunately it throws some runtime errors 

```
2025-05-28 13:23:36.098 [info] [Error - 13:23:36] The file does not exist in the workspace.
2025-05-28 13:23:36.098 [info] [Error - 13:23:36] The file does not exist in the workspace.
2025-05-28 13:23:36.100 [info] [Error - 13:23:36] The file does not exist in the workspace.
2025-05-28 13:23:36.100 [info] [Error - 13:23:36] The file does not exist in the workspace.
2025-05-28 13:23:36.101 [info] [Error - 13:23:36] The file does not exist in the workspace.
2025-05-28 13:23:36.101 [info] [Error - 13:23:36] The file does not exist in the workspace.
2025-05-28 13:23:36.101 [info] [Error - 13:23:36] The file does not exist in the workspace.
```